### ISSUE TITLE
 Assertion Failed: <id@serializer:patient::ember1289> has opted into the new serializer API and expects the <id@serializer:-active-model::ember1290> it collaborates with to also support the new serializer API by setting its `isNewSerializerAPI` property to true.

### ISSUE THREAD

**Issue by jcope2013** [ember-data/active-model-adapter#13]
having difficulty with getting this active-model-adapter to work, these are in my console

```
Assertion Failed: <id@serializer:patient::ember1289> has opted into the new serializer API and expects the <id@serializer:-active-model::ember1290> it collaborates with to also support the new serializer API by setting its `isNewSerializerAPI` property to true.
```

```
Your custom serializer uses the old version of the Serializer API, with `extract` hooks. Please upgrade your serializers to the new Serializer API using `normalizeResponse` hooks instead.
```

even though my code looks like this

app/serializers/patient.js

```
import { ActiveModelSerializer } from 'active-model-adapter';

export default ActiveModelSerializer.extend({
  isNewSerializerAPI: true,

  keyForAttribute: function(attr, method) {
    if (attr === 'patientPlans') {
      return 'patient_plan_ids';
    }
    return this._super(attr);
  }
});
```

app/adapters/application.js

```

import ENV from '../config/environment';
import ActiveModelAdapter from 'active-model-adapter';

export default ActiveModelAdapter.extend({
  namespace: ENV.namespace,
  host: ENV.host,
  ajax: function(url, method, hash) {
    hash = hash || {}; // hash may be undefined
    hash.crossDomain = true;
    hash.xhrFields = {withCredentials: true};
    return this._super(url, method, hash);
  }
});

```

even switching all them back to DS.ActiveModelAdapter and DS.ActiveModelSerializer is giving me difficulty with the package still in my app, its only when i remove the package from my app that my app goes back to working fine

Ember 1.13.2
Ember Data 1.13.4
Ember CLI 0.2.7
active-model-adapter 1.13.2

**Comment by wecc** [comment]
I think this should have been solved by #8, might need a new release.

**Comment by fivetanley** [comment]
I just released 1.13.3. Can you try that? Thanks.

**Comment by jcope2013** [comment]
@wecc @fivetanley still seeing same issue with 1.13.3

**Comment by poteto** [comment]
@jcope2013 As a workaround, you can `reopen` the `ActiveModelSerializer` prior to extending it, then moving the flag there:

```js
import { ActiveModelSerializer } from 'active-model-adapter';

ActiveModelSerializer.reopen({
  isNewSerializerAPI: true
});

export default ActiveModelSerializer.extend({
  // ...
});
```

**Comment by jcope2013** [comment]
@poteto are you doing that in all of your serializers, application serializer or just any random serializer but only one time? thanks

**Comment by blimmer** [comment]
I ran into this problem when unit-testing my serializer. I think because the initializer doesn't run, the `isNewSerializerAPI` flag does not get set, which blows things up.

**Comment by fivetanley** [comment]
I think this got closed by #32. Feel free to reopen if I'm wrong.
